extension/runtime/src/main/java/org/iris_events/producer/EventProducer.java

```diff
@@ -146,6 +165,60 @@ public void send(final Object message, final String userId, final boolean propag
         doSend(message, userId, propagate);
     }
 
+    public <T> T sendRpcRequest(final Object message, Class<T> responseType) {
```

Comment:
rename to `sendAndRecieve()`
also please add overloaded method with timeout (how much to wait for response)

Reply:
Renamed and added the overloaded method.

Reply:
For the scenario described above (not defining the rpcResponse and still calling this method), that won't work as IRIS uses the `rpcResponse` to set the `replyTo` parameter. Also supporting this will be tricky, as we'll need to scan all calls of this `sendAndReceive` method, because we have to propagate all needed data also through asyncapi into generated models. I'm sure there is a way of doing it, but with the current implementation where the consumer is the boss of defining all RPC rabbit infrastructure it'll be really tricky. We can still try, but as you said I would do it in the next iteration.